src/index.js

```diff
@@ -247,36 +279,34 @@ async function watch({ seq }) {
       return_docs: false, // eslint-disable-line camelcase
     });
 
-    const q = queue((change, done) => {
-      saveDocs({ docs: [change], index: mainIndex })
-        .then(() => infoChange(change.seq, 1, '🛰'))
-        .then(() =>
-          stateManager.save({
-            seq: change.seq,
-          })
-        )
-        .then(stateManager.get)
-        .then(({ bootstrapLastDone }) => {
-          const now = Date.now();
-          const lastBootstrapped = new Date(bootstrapLastDone);
-          // when the process is running longer than a certain time
-          // we want to start over and get all info again
-          // we do this by exiting and letting Heroku start over
-          if (now - lastBootstrapped > c.timeToRedoBootstrap) {
-            return stateManager
-              .set({
-                seq: 0,
-                bootstrapDone: false,
-              })
-              .then(() => {
-                process.exit(0); // eslint-disable-line no-process-exit
-              });
-          }
-
-          return null;
-        })
-        .then(done)
-        .catch(done);
+    const q = queue(async change => {
+      datadog.increment('packages');
+
+      try {
+        await saveDocs({ docs: [change], index: mainIndex });
+        await infoChange(change.seq, 1, '🛰');
+        await stateManager.save({
+          seq: change.seq,
+        });
+        const { bootstrapLastDone } = await stateManager.get();
+
+        const now = Date.now();
+        const lastBootstrapped = new Date(bootstrapLastDone);
+        // when the process is running longer than a certain time
+        // we want to start over and get all info again
+        // we do this by exiting and letting Heroku start over
+        if (now - lastBootstrapped > c.timeToRedoBootstrap) {
+          await stateManager.set({
+            seq: 0,
+            bootstrapDone: false,
+          });
+          process.exit(0); // eslint-disable-line no-process-exit
+        }
+
+        return null;
+      } catch (e) {
+        return e;
```

Comment:
is this the same as "return done"?

Reply:
`done` is not defined when using async/await, it should be similar. All my tests seems to proove so, but your tests will be mandatory on this.